### PR TITLE
Enable egress service e2es for ipv6

### PR DESCRIPTION
**Pull request by kyrtapz**

### CODE REVIEW

**Comment by kyrtapz** [comment]
`e2e (control-plane, noHA, local, ipv6, noSnatGW, 1br, ic-single-node-zones)` failed with an unrelated issue:
```
Summarizing 1 Failure:
  [FAIL] Service Hairpin SNAT [It] Should ensure service hairpin traffic is NOT SNATed to hairpin masquerade IP; GR LB
  /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/service.go:958

Ran 35 of 262 Specs in 809.004 seconds
FAIL! -- 34 Passed | 1 Failed | 1 Pending | 226 Skipped
```
https://github.com/ovn-org/ovn-kubernetes/actions/runs/8152636379/job/22283136604?pr=4200

**Comment by kyrtapz** [comment]
` e2e (multi-homing, noHA, local, dualstack, SnatGW, 1br, ic-single-node-zones)` doesn't look related either:
```
• [FAILED] [244.978 seconds]
Multi Homing multiple pods connected to the same OVN-K secondary network localnet OVN-K secondary network with a service running on the underlay when a policy is provisioned [BeforeEach] can communicate over a localnet secondary network from pod to gw ingress denyall, egress allow all, ingress policy should have no impact on egress
  [BeforeEach] /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/multihoming.go:806
  [It] /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/multihoming.go:855

  [FAILED] Timed out after 120.001s.
  Expected
      <v1.PodPhase>: Pending
  to equal
      <v1.PodPhase>: Running
  In [BeforeEach] at: /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/multihoming.go:813 @ 03/05/24 10:06:15.376

  There were additional failures detected.  To view them in detail run ginkgo -vv
```
https://github.com/ovn-org/ovn-kubernetes/actions/runs/8152636379/job/22283138542?pr=4200

File: contrib/kind.sh [hunk 1]
```diff
@@ -1064,28 +1080,44 @@ install_metallb() {
...
-  client_subnet=$(docker network inspect clientnet -f '{{range .IPAM.Config}}{{.Subnet}}{{end}}')
-  echo "${client_subnet}"
+  kind_network_v4=$(docker inspect -f '{{index .NetworkSettings.Networks "kind" "IPAddress"}}' frr)
+  echo "FRR kind network IPv4: ${kind_network_v4}"
+  kind_network_v6=$(docker inspect -f '{{index .NetworkSettings.Networks "kind" "GlobalIPv6Address"}}' frr)
+  echo "FRR kind network IPv6: ${kind_network_v6}"
+  local client_network_v4 client_network_v6
+  client_network_v4=$(docker inspect -f '{{index .NetworkSettings.Networks "clientnet" "IPAddress"}}' frr)
+  echo "FRR client network IPv4: ${client_network_v4}"
+  client_network_v6=$(docker inspect -f '{{index .NetworkSettings.Networks "clientnet" "GlobalIPv6Address"}}' frr)
```
Comment (tssurya) [inline contrib/kind.sh:1090]: @kyrtapz : Is this interim error expected and ok:?
```
Status: Downloaded newer image for quay.io/frrouting/frr:9.0.2
for frr in $(docker ps -a -f name=frr --format {{.Names}}) ; do     docker rm -f $frr ; done
docker run -d --privileged --network kind --rm --ulimit core=-1 --name frr --volume /tmp/tmp.Bkxk1K05En/frr-volume:/etc/frr quay.io/frrouting/frr:9.0.2
505d34ffa6af7ba669c28dc9c5aa594dc0819a5c41a5cf447629998c010eeed9
docker inspect -f "{{ range .NetworkSettings.Networks}}{{.GlobalIPv6Address}}{{end}}" frr
fc00:f853:ccd:e793::5
Error from server (InternalError): error when creating "/tmp/metallb/jo2MP9/metallb/dev-env/bgp/config.yaml": Internal error occurred: failed calling webhook "bgppeersvalidationwebhook.metallb.io": failed to call webhook: Post "[https://metallb-webhook-service.metallb-system.svc:443/validate-metallb-io-v1beta2-bgppeer?timeout=10s](https://metallb-webhook-service.metallb-system.svc/validate-metallb-io-v1beta2-bgppeer?timeout=10s)": context deadline exceeded
bgpadvertisement.metallb.io/empty created
Error from server (InternalError): error when creating "/tmp/metallb/jo2MP9/metallb/dev-env/bgp/config.yaml": Internal error occurred: failed calling webhook "ipaddresspoolvalidationwebhook.metallb.io": failed to call webhook: Post "[https://metallb-webhook-service.metallb-system.svc:443/validate-metallb-io-v1beta1-ipaddresspool?timeout=10s](https://metallb-webhook-service.metallb-system.svc/validate-metallb-io-v1beta1-ipaddresspool?timeout=10s)": context deadline exceeded
Sleeping for 2s
bgppeer.metallb.io/peer created
ipaddresspool.metallb.io/dev-env-bgp created
bgpadvertisement.metallb.io/empty unchanged
+ docker network create --subnet=172.22.0.0/16 --ipv6 --subnet=fc00:f853:ccd:e792::/64 --driver bridge clientnet
fb3b3e2f56c2593f8b737f29a0944c600e9a8880d7387dabcf15b23a6535870f
+ docker network connect clientnet frr
+ docker run --cap-add NET_ADMIN --user 0 -d --network clientnet --rm --name lbclient quay.io/itssurya/dev-images:metallb-lbservice
Unable to find image 'quay.io/itssurya/dev-images:metallb-lbservice' locally
metallb-lbservice: Pulling from itssurya/dev-images
```

Comment (tssurya) [reply contrib/kind.sh:1090]: nit: we could avoid trying to get the v6 information on the v4 lanes but I am ok if you say "let's let it be" :)
```
+ local client_network_v4 client_network_v6
++ docker inspect -f '{{index .NetworkSettings.Networks "clientnet" "IPAddress"}}' frr
+ client_network_v4=172.22.0.2
FRR client network IPv4: 172.22.0.2
+ echo 'FRR client network IPv4: 172.22.0.2'
++ docker inspect -f '{{index .NetworkSettings.Networks "clientnet" "GlobalIPv6Address"}}' frr
+ client_network_v6=
FRR client network IPv6: 
+ echo 'FRR client network IPv6: '
+ local client_subnets
++ docker network inspect clientnet -f '{{range .IPAM.Config}}{{.Subnet}}#{{end}}'
+ client_subnets=172.22.0.0/16#
172.22.0.0/16#
+ echo 172.22.0.0/16#
+ local client_subnets_v4 client_subnets_v6
++ echo 172.22.0.0/16#
++ cut -d '#' -f 1
+ client_subnets_v4=172.22.0.0/16
+ echo 'client subnet IPv4: 172.22.0.0/16'
client subnet IPv4: 172.22.0.0/16
++ echo 172.22.0.0/16#
++ cut -d '#' -f 2
+ client_subnets_v6=
+ echo 'client subnet IPv6: '
client subnet IPv6: 
++ kind get nodes --name ovn
```

File: test/scripts/e2e-cp.sh
Comment (tssurya) [inline test/scripts/e2e-cp.sh:1]: nit: fix comment on L17/18

**Review by tssurya — COMMENTED** [review]
@kyrtapz : you ran the ESVC tests locally right? Because I tried to check the logs and nowadays we probably don't run ginkgo in verbose mode? so there are no logs, then I tried to download the KIND cluster logs from v6 run and there I tried to grep for `test-egress-service` and found nothing => maybe another test ends up restarting ovnkube-node so we loose logs dunno, but confirming we have indeed double verified these tests are running?

**Comment by kyrtapz** [comment]
> @kyrtapz : you ran the ESVC tests locally right? Because I tried to check the logs and nowadays we probably don't run ginkgo in verbose mode?

I ran the tests locally but it turns out lgw had `Services` in the skipped tests for lgw with matched `Egress Services`. I removed it.

**Review by tssurya — DISMISSED** [review]
/lgtm
will merge when CI is greeeeeeen
thanks Pat

**Comment by tssurya** [comment]
https://github.com/ovn-org/ovn-kubernetes/actions/runs/8192617883/job/22405564518?pr=4200 flake unrelated we already have a CI for that: https://github.com/ovn-org/ovn-kubernetes/issues/4144

File: test/e2e/egress_services.go [hunk 2]
```diff
@@ -1160,7 +1160,9 @@ spec:
 				gomega.Consistently(func() error {
 					for _, pod := range append(net1.createdPods, net2.createdPods...) {
 						err := curlAgnHostClientIPFromPod(f.Namespace.Name, pod, "", dst, podHTTPPort)
-						if err != nil && strings.Contains(err.Error(), "exit code 28") {
+						if err != nil && (strings.Contains(err.Error(), fmt.Sprintf("exit code 28")) ||
+							// github runners don't have any routes for IPv6, so we get CURLE_COULDNT_CONNECT
```
Comment (tssurya) [inline test/e2e/egress_services.go:1164]: hehe the level of detail one has to go to! Can you provide source of how you found this? (both the github runner limitation and the source code 7 versus 28)

**Review by tssurya — APPROVED** [review]
/lgtm
thanks

**Comment by tssurya** [comment]
double checked that esvc-es are running on lgw mode.

**Pull request merged**